src/gfx/colorers/CarbonColorer.js

```diff
@@ -0,0 +1,51 @@
+import Colorer from './Colorer';
+import selectors from '../../chem/selectors';
+import Atom from '../../chem/Atom';
+
+function scaleColor(c, factor) {
+  var
+    r1 = (c >> 16) & 0xff,
+    g1 = (c >> 8) & 0xff,
+    b1 = c & 0xff;
+  var
+    r = factor * r1,
+    g = factor * g1,
+    b = factor * b1;
+  return (r << 16) | (g << 8) | b;
+}
+
+/**
+ * Create new colorer.
+ *
+ * @param {object=} opts - Options to override defaults with. See {@link Colorer}.
+ *
+ * @exports ConditionalColorer
+ * @augments Colorer
+ * @constructor
+ * @classdesc Bicolor coloring algorithm based on a selector string used as a condition.
+ */
+class CarbonColorer extends Colorer {
+  static id = 'CB';
+
+  constructor(opts) {
+    super(opts);
+    const parsed = selectors.parse(this.opts.subset);
+    this._subsetCached = parsed.error ? selectors.none() : parsed.selector;
+  }
+
+  getAtomColor(atom, _complex) {
+    var colorCarbon = this.opts.color;
+    var colorNotCarbon = scaleColor(colorCarbon, this.opts.factor);
+    return (atom.flags === Atom.Flags.CARBON) ? colorCarbon : colorNotCarbon;
```

Comment:
We should check that this particular flag is on, ignoring other bits if there are any (like [here](https://github.com/epam/miew/blob/master/src/gfx/colorers/SecondaryStructureColorer.js#L16)).